Neos.Neos/Classes/FrontendRouting/CatchUpHook/DocumentUriPathProjectionHookForRouteCacheFlushing.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Neos\Neos\FrontendRouting\CatchUpHook;
+
+use Neos\ContentRepository\Core\Projection\CatchUpHookInterface;
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\EventStore\EventInterface;
+use Neos\EventStore\Model\EventEnvelope;
+use Neos\ContentRepository\Core\Feature\NodeModification\Event\NodePropertiesWereSet;
+use Neos\ContentRepository\Core\Feature\NodeMove\Event\NodeAggregateWasMoved;
+use Neos\ContentRepository\Core\Feature\NodeRemoval\Event\NodeAggregateWasRemoved;
+use Neos\RedirectHandler\NeosAdapter\Service\NodeRedirectService;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\Neos\FrontendRouting\Projection\DocumentUriPathFinder;
+use Neos\ContentRepository\Core\Feature\NodeMove\Dto\CoverageNodeMoveMapping;
+use Neos\Neos\FrontendRouting\Projection\DocumentNodeInfo;
+use Neos\Neos\FrontendRouting\Exception\NodeNotFoundException;
+use Neos\Neos\FrontendRouting\NodeAddress;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\ContentRepository\Core\Feature\NodeDisabling\Event\NodeAggregateWasDisabled;
+use Neos\Neos\Routing\Cache\RouteCacheFlusher;
+use Neos\Flow\Mvc\Routing\RouterCachingService;
+
+final class DocumentUriPathProjectionHookForRouteCacheFlushing implements CatchUpHookInterface
```

Comment:
Just a nitpick, but this is a veery long class name, what about just `RouterCacheHook` and a short class doc comment?

Reply:
Sure, I was inspired by `GraphProjectorCatchUpHookForCacheFlushing` but I don't insist on anything.